### PR TITLE
XS✔ ◾ PR Metrics v1.2.4: Unifying access token variable names for GitHub & Azure DevOps

### PR DESCRIPTION
**Pull request by muiriswoulfe**
## Summary

This release bumps the version number of PR Metrics to 1.2.4 and makes the following changes:

- Unifying the access token variable names for GitHub and Azure DevOps as `SYSTEM_ACCESSTOKEN`.
- Simplifying the error messages displayed when an access token has insufficient access.
- Better handling insufficient access in Azure DevOps scenarios with a clearer error message.
- Addition of new manual testing instructions for validating this case.
- Updates to dependencies.

## Testing

Unit test coverage was maintained and expanded to cover the new scenarios. All unit tests are passing reliably.

Manual testing instructions were expanded and followed to ensure no regressions occurred.

### CODE REVIEW

**Review by AndreyTretyak — DISMISSED** [review]

File: PipelinesTasks/PRMetrics/README.md [hunk 1]
```diff
@@ -83,8 +83,9 @@ definition.
 
 For GitHub repos, you will need to create a PAT according to the instructions
 [here][githubpat]. The scope should be 'repos'. The resulting PAT should then be
-added to your repo as a secret with the name `PRMetrics.PAT` using the
-instructions [here][githubsecret].
+added to your repo as a secret with the name `System.AccessToken` using the
+instructions [here][githubsecret]  and mapped to `System.AccessToken` within the
```
Comment (dimeehan) [inline PipelinesTasks/PRMetrics/README.md:87]: nit: extra space before "and"

File: PipelinesTasks/PRMetrics/src/LICENSE.txt [hunk 2]
```diff
@@ -1368,7 +1368,7 @@ THE SOFTWARE.
```
Comment (dimeehan) [inline PipelinesTasks/PRMetrics/src/LICENSE.txt:1368]: nit: not apart of this PR, but how come the copyright comment is in twice around line 49/50, is that on purpose?

Comment (muiriswoulfe) [reply PipelinesTasks/PRMetrics/src/LICENSE.txt:1368]: I realise this is strange, but the file is autogenerated and it's seemingly correct. The second instance of the copyright statement is coming from <https://github.com/microsoft/tslib/blob/main/LICENSE.txt>. Not sure where the first comes from, but I assume it is some repo metadata. I guess they're both the same in this case, causing the repetition.

File: PipelinesTasks/PRMetrics/README.md (hunk 1 above)
Comment (muiriswoulfe) [reply PipelinesTasks/PRMetrics/README.md:87]: Good catch, thanks!

File: PipelinesTasks/PRMetrics/src/task/tests/repos/gitHubReposInvoker.spec.ts [hunk 3]
```diff
@@ -437,15 +439,15 @@ describe('gitHubReposInvoker.ts', function (): void {
     })
 
     it('should return a string when the token does not exist', (): void => {
-      // Arrange
-      when(taskLibWrapper.getVariable('PRMetrics.PAT')).thenReturn(undefined)
+      // Arrange#
```
Comment (dimeehan) [inline PipelinesTasks/PRMetrics/src/task/tests/repos/gitHubReposInvoker.spec.ts:442]: nit: is the # added here by mistake?

Comment (muiriswoulfe) [reply PipelinesTasks/PRMetrics/src/task/tests/repos/gitHubReposInvoker.spec.ts:442]: Yes, another good catch :)

File: PipelinesTasks/PRMetrics/src/task/src/repos/baseReposInvoker.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  public abstract updateComment(content: string | null, status: CommentThreadStatus | null, commentThreadId: number): Promise<void>
+
+  public abstract deleteCommentThread(commentThreadId: number): Promise<void>
+
+  /**
+   * Invokes an API call, augmenting any errors that may be thrown due to insufficient access.
+   * @typeParam TResponse The type of the response from the API call.
+   * @param action The action defining the API call to invoke.
+   * @param errorMessage The error message to insert if a caught error is due to insufficient access.
+   * @returns A promise containing the the response from the API call.
```
Comment (dimeehan) [inline PipelinesTasks/PRMetrics/src/task/src/repos/baseReposInvoker.ts:33]: the the :-)

**Review by dimeehan — APPROVED** [review]

**Review by dimeehan — APPROVED** [review]

**Review by AndreyTretyak — APPROVED** [review]

**Pull request merged**